hazelcast/src/main/java/com/hazelcast/internal/partition/MigrationEventHandler.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.partition;
+
+import com.hazelcast.partition.MigrationListener;
+import com.hazelcast.partition.MigrationState;
+import com.hazelcast.partition.ReplicaMigrationEvent;
+
+public final class MigrationEventHandler {
+
+    public static final int MIGRATION_STARTED_PARTITION_ID = -1;
+    public static final int MIGRATION_FINISHED_PARTITION_ID = -2;
+
+    private final MigrationListener migrationListener;
+
+    public MigrationEventHandler(MigrationListener migrationListener) {
+        this.migrationListener = migrationListener;
+    }
+
+    public void handleMigrationEvent(MigrationState state, MigrationEventType type) {
+        switch (type) {
+            case MIGRATION_STARTED:
+                migrationListener.migrationStarted(state);
+                break;
+            case MIGRATION_FINISHED:
+                migrationListener.migrationFinished(state);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid event type: " + type);
```

Comment:
Leaving the default state empty make much more sense if you think about future extensions. Let's say this client connected to a new server, which has a new migration event type. Old clients should not throw an exception or log something aggressively because it is expected and old clients should just ignore it.